web/src/composebox_typeahead.ts

```diff
@@ -960,16 +953,49 @@ export function get_candidates(
                 }
 
                 const stream_id = stream_data.get_stream_id(stream_name);
-                const topic_list = topics_seen_for(stream_id);
+                // We always show topic suggestions after the user types a stream,
+                // and let them pick between just showing the stream (the first option, when nothing follows ">")
+                // or adding a topic.
+                const topic_list = current_token.endsWith(">")
+                    ? [stream_name, ...topics_seen_for(stream_id)]
+                    : topics_seen_for(stream_id);
+
                 if (should_show_custom_query(token, topic_list)) {
                     topic_list.push(token);
                 }
                 const matcher = get_topic_matcher(token);
                 const matches = topic_list.filter((item) => matcher(item));
-                const matches_list: TopicSuggestion[] = matches.map((topic) => ({
-                    topic,
-                    type: "topic_list",
-                }));
+                const matches_list: TopicSuggestion[] = matches.map((topic) => {
```

Comment:
You can simplify this. Including my above suggestion to always have stream data present:

```js
const stream = stream_data.get_sub_by_name(stream_name);
assert(stream !== undefined);
return {
    topic,
    type: "topic_list",
    is_stream_only: topic === stream_name,
    streamData: {...stream, type: "stream"},
};
```